### PR TITLE
Fix channel num for depthwise conv3d

### PR DESCRIPTION
**Pull request by ShuaiShao93**

### CODE REVIEW

**Comment by fatcat-z** [comment]
Thanks for your contribution.

Could you please:
1. Open a new issue to describe this problem and link it to this PR?
2. Add a test for this change?

**Comment by ShuaiShao93** [comment]
> Thanks for your contribution.
> 
> Could you please:
> 
> 1. Open a new issue to describe this problem and link it to this PR?
> 2. Add a test for this change?

Sure. I'm not verify familiar with the codebase, can you suggest where to add the unit tests?

**Comment by fatcat-z** [comment]
> > Thanks for your contribution.
> > Could you please:
> > 
> > 1. Open a new issue to describe this problem and link it to this PR?
> > 2. Add a test for this change?
> 
> Sure. I'm not verify familiar with the codebase, can you suggest where to add the unit tests?

You can add it [here](https://github.com/onnx/tensorflow-onnx/blob/01520291227d46c48615300b0a73436dbe3c6610/tests/test_backend.py#L464C4-L464C4).

**Comment by ShuaiShao93** [comment]
> > > Thanks for your contribution.
> > > Could you please:
> > > 
> > > 1. Open a new issue to describe this problem and link it to this PR?
> > > 2. Add a test for this change?
> > 
> > 
> > Sure. I'm not verify familiar with the codebase, can you suggest where to add the unit tests?
> 
> You can add it [here](https://github.com/onnx/tensorflow-onnx/blob/01520291227d46c48615300b0a73436dbe3c6610/tests/test_backend.py#L464C4-L464C4).

Thanks! I added a test but it failed with wrong results. Can this be a bug with the onnx runtime?

**Comment by fatcat-z** [comment]
> > > > Thanks for your contribution.
> > > > Could you please:
> > > > 
> > > > 1. Open a new issue to describe this problem and link it to this PR?
> > > > 2. Add a test for this change?
> > > 
> > > 
> > > Sure. I'm not verify familiar with the codebase, can you suggest where to add the unit tests?
> > 
> > 
> > You can add it [here](https://github.com/onnx/tensorflow-onnx/blob/01520291227d46c48615300b0a73436dbe3c6610/tests/test_backend.py#L464C4-L464C4).
> 
> Thanks! I added a test but it failed with wrong results. Can this be a bug with the onnx runtime?

What do you mean by "wrong results"? The result of ORT is different than tf? Is it possible for you to export the onnx graph of this test to check if it is correct?

**Comment by ShuaiShao93** [comment]
> > > > > Thanks for your contribution.
> > > > > Could you please:
> > > > > 
> > > > > 1. Open a new issue to describe this problem and link it to this PR?
> > > > > 2. Add a test for this change?
> > > > 
> > > > 
> > > > Sure. I'm not verify familiar with the codebase, can you suggest where to add the unit tests?
> > > 
> > > 
> > > You can add it [here](https://github.com/onnx/tensorflow-onnx/blob/01520291227d46c48615300b0a73436dbe3c6610/tests/test_backend.py#L464C4-L464C4).
> > 
> > 
> > Thanks! I added a test but it failed with wrong results. Can this be a bug with the onnx runtime?
> 
> What do you mean by "wrong results"? The result of ORT is different than tf? Is it possible for you to export the onnx graph of this test to check if it is correct?

Ah it doesn't work for all the TF versions. After adding tf version constraints, it works now

**Comment by ShuaiShao93** [comment]
Also filed issue: https://github.com/onnx/tensorflow-onnx/issues/2255

File: tests/test_backend.py [hunk 1]
```diff
@@ -695,6 +695,23 @@ def func(x, y):
             return tf.identity(conv, name=_TFOUTPUT)
         self._run_test_case(func, [_OUTPUT], {_INPUT: x_val, _INPUT1: x_val}, rtol=0.08)
 
+    @check_tf_min_version("2.11")
+    @check_tf_max_version("2.12")
```
Comment (fatcat-z) [inline tests/test_backend.py:699]: Does it mean this test only works for 2.11 and 2.12 versions? Did you try it on 2.13?

Could you please describe what kind of issues you met when tf version < 2.11?

Comment (ShuaiShao93) [reply tests/test_backend.py]: Actually, 2.13 works, but old TF versions don't work. PTAL again!

Comment (ShuaiShao93) [reply tests/test_backend.py]: Hmm, for some reason, `TF2ONNX_TEST_OPSET=18 python -m unittest test_backend.BackendTests.test_depthwise_conv3d` can pass locally after `pip install tensorflow==2.13.0`, but it's still failing on CI.

Comment (fatcat-z) [reply tests/test_backend.py]: In CI, it failed from opset 15 to opset 18, not only opset 18. Looks like the failures are relative to 2.13.0. Could you please check the tf nodes got during executing tflist_to_onnx() method to see what are differences between tensorflow 2.8.0 and 2.13.0?

**Pull request closed without merging**